### PR TITLE
Feat/15536366 reveal flow UI

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15536366

image
image
image

### CODE REVIEW

File: src/components/v5/shared/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -8,8 +8,9 @@ const ProgressBar: FC<ProgressBarProps> = ({
   isTall,
   additionalText,
   threshold,
+  max = 100,
 }) => {
-  if (progress > 100 || progress < 0) {
+  if (progress > max || progress < 0) {
     throw new Error('Progress bar value must be between 0 and 100');
```
Comment (TomaszChmielPagepro) [inline src/components/v5/shared/ProgressBar/ProgressBar.tsx:14]: `between 0 and ${max}`

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/MotionSimplePayment.tsx [hunk 2]
```diff
@@ -97,12 +98,28 @@ const MotionSimplePayment: FC<MotionSimplePaymentProps> = ({
             },
             {
               key: MotionState.Reveal,
-              content: <RevealStep />,
+              content: (
+                <RevealStep
+                  motionData={motionData as ColonyMotion}
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/MotionSimplePayment.tsx:103]: try to aviod casting

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/MotionSimplePayment.tsx [hunk 3]
```diff
@@ -97,12 +98,28 @@ const MotionSimplePayment: FC<MotionSimplePaymentProps> = ({
...
+                  startPollingAction={startPollingForAction}
+                  stopPollingAction={stopPollingForAction}
+                  transactionId={transactionId}
+                />
+              ),
               heading: {
                 label: formatText({ id: 'motion.reveal.label' }) || '',
+                decor:
+                  motionStateEnum === MotionState.Reveal && motionStakes ? (
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/MotionSimplePayment.tsx:112]: Can't it just be like this:  motionStateEnum === MotionState.Reveal && motionStakes && ( ... ) ?

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/RevealStep.tsx [hunk 4]
```diff
@@ -1,10 +1,155 @@
...
+                  <MotionBadge status={isSupportVote ? 'support' : 'oppose'} />
+                </div>
+                <div className="flex items-center justify-between gap-2">
+                  <span className="text-sm text-gray-600">
+                    {formatText({ id: 'motion.revealStep.rewards' })}
+                  </span>
+                  <span className="text-sm">
+                    <Numeral
+                      value={voterReward || '0'}
+                      decimals={nativeToken?.decimals}
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/RevealStep.tsx:98]: please use destructuring assignment

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/RevealStep.tsx [hunk 5]
```diff
@@ -1,10 +1,155 @@
...
+                      suffix={nativeToken?.symbol}
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/RevealStep.tsx:99]: the same

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/hooks.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+  const { vote, hasUserVoted, userVoteRevealed, setUserVoteRevealed } =
+    useRevealWidgetUpdate(voterRecord, stopPollingAction);
+  const transform = mapPayload(
+    () =>
+      ({
+        colonyAddress: colony?.colonyAddress,
+        userAddress: user?.walletAddress ?? '',
+        motionId: BigNumber.from(motionId),
+      } as RevealMotionPayload),
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/hooks.ts:46]: please avoid custing

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/hooks.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  );
+
+  const handleSuccess: OnSuccess<Record<string, any>> = (_, { reset }) => {
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/hooks.ts:49]: please avoid using any if it possible

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/hooks.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  };
+
+  const voters = voterRecord.map((voter) => ({
+    address: voter.address,
+    hasRevealed: voter.vote !== null,
+  }));
+  const userVote =
+    hasUserVoted && (vote || getLocalStorageVoteValue(transactionId));
+  const isSupportVote = userVote === MotionVote.Yay;
+  const revealProgress = voterRecord.reduce(
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/hooks.ts:62]: maybe will be better to use here useMemo?

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationItem/RevealInformationItem.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React, { FC } from 'react';
+import { RevealInformationListItem } from './types';
+import { useUserByAddress } from '~hooks';
+import AvatarUser from '~v5/shared/AvatarUser';
+import Icon from '~shared/Icon';
+
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationItem/RevealInformationItem.tsx:6]: missing displayName

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationList.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import React, { FC, useState } from 'react';
+
+import RevealInformationItem from './RevealInformationItem';
+
+import { RevealInformationListProps } from './types';
+import { TextButton } from '~v5/shared/Button';
+import { formatText } from '~utils/intl';
+
+const RevealInformationList: FC<RevealInformationListProps> = ({ items }) => {
+  const [itemsLength, setItemsLength] = useState(6);
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationList.tsx:10]: please move 6 number to const file

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationList.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+  const itemsToShow = items.slice(0, itemsLength);
+  const shouldShowButton = items.length > 6 && itemsLength < items.length;
+
+  return (
+    <div>
+      <ul className="pt-6">
+        {itemsToShow.map(({ address, hasRevealed }) => (
+          <li key={address} className="mb-3 last:mb-0">
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationList.tsx:19]: Is it possible to use other value as a key?

Comment (CzarekDryl) [reply src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/RevealStep/partials/RevealInformationList.tsx:19]: There is only address and vote inside voters object, but addresses can't be the same for users so I guess it's not bad

File: src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/MotionSimplePayment.tsx (hunk 3 above)
Comment (CzarekDryl) [reply src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/MotionSimplePayment.tsx:111]: Decor is types as optional JSX.Element so it needs undefined if there is no content

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**